dotCMS/src/main/java/com/dotmarketing/util/ImportStarterUtil.java

```diff
@@ -0,0 +1,1148 @@
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(Layout.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+
+        
+        
+        for (File file : contains(PortletsLayouts.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+
+        
+        for (File file : endsWith("LogsMappers.xml")) {
+            doXMLFileImport(file);
+        }
+        
+        for (File file : contains(LayoutsRoles.class.getCanonicalName())) {
+            doXMLFileImport(file);
+        }
+        
+        
+        
+        for (File file : contains(Language.class.getCanonicalName() + "_")) {
+            doXMLFileImport(file);
+        }
+        /*
+         * Because of the parent check we do at db we need to import folder identifiers first but for the
+         * same reason we need to sort them first by parent_path
+         */
+        final List<Identifier> folderIdents = new ArrayList<Identifier>();
+        final List<Identifier> otherIdents = new ArrayList<Identifier>();
+
+        List<File> identifiers = contains(Identifier.class.getCanonicalName());
+        // collecting all folder identifiers
+        for (File ff : identifiers) {
+            List<Identifier> idents = (List<Identifier>) BundlerUtil.xmlToObject(ff);
+            for (Identifier ident : idents) {
+                if (ident.getAssetType().equals("folder")) {
+                    folderIdents.add(ident);
+                }else {
+                    otherIdents.add(ident);
+                }
+            }
+        }
+
+        // sorting folder identifiers by parent path in order to pass parent check
+        Collections.sort(folderIdents, new Comparator<Identifier>() {
+            public int compare(Identifier o1, Identifier o2) {
+                return o1.getParentPath().compareTo(o2.getParentPath());
+            }
+        });
+
+        // saving folder identifiers
+        for (Identifier ident : folderIdents) {
+            if (!SYSTEM_FOLDER_PATH.equals(ident.getParentPath())) {
+                ident.setParentPath(ident.getParentPath());
+                ident.setAssetName(ident.getAssetName());
+            }
+            Logger.info(this, "Importing folder path " + ident.getParentPath() + ident.getAssetName());
+            APILocator.getIdentifierAPI().save(ident);
+        }
+
+        // now we need to save all remaining identifiers (folders already added)
+        for (final Identifier ident : otherIdents) {
+            APILocator.getIdentifierAPI().save(ident);
+        }
+
+
+
+        // we store here defaultFileType for every folder
+        // because of mutual folder <--> structure dependency
+        final Map<String, String> fileTypesInodes = new HashMap<String, String>();
+
+        // now we can import folders
+        for (File ff : contains(Folder.class.getCanonicalName())) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'ff' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)